packages/material-ui/src/InputBase/InputBase.test.js

```diff
@@ -182,6 +182,21 @@ describe('<InputBase />', () => {
       assert.strictEqual(typeof injectedProps.onBlur, 'function');
       assert.strictEqual(typeof injectedProps.onFocus, 'function');
     });
+
+    it('throws on change if `inputRef` isnt forwarded', () => {
+      // eslint-disable-next-line react/prop-types
+      const BadInputComponent = React.forwardRef(({ inputRef, ...props }, ref) => {
+        // `inputRef` belongs into `ref`
+        return <input {...props} ref={ref} />;
+      });
+
+      const wrapper = mount(<InputBase inputComponent={BadInputComponent} />);
+      assert.throws(
+        // simulating e.g. react-stripe-elements that hides the event target
+        () => wrapper.find('input').simulate('change', { target: null }),
```

Comment:
aside: no idea why this simulate is StrictMode compatible but `fireEvent.change` doesn't even register. Probably missing something obvious about `change` events.